packages/mui-lab/package.json

```diff
@@ -83,7 +83,7 @@
     "@types/luxon": "^1.27.1",
     "date-fns": "^2.25.0",
     "dayjs": "^1.10.7",
-    "luxon": "^1.28.0",
+    "luxon": "^1.21.3 || ^2.x",
```

Comment:
Hoisted the peer dependency range from https://unpkg.com/@date-io/luxon@2.11.1/package.json

Reply:
Should've changed the `peerDependencies`. Actual fix is in https://github.com/mui-org/material-ui/pull/29669